h3/src/params.rs

```diff
@@ -0,0 +1,45 @@
+//! HTTP/3 connection parameters
+
+/// HTTP/3 connection parameters builder
+pub struct Params {
+    pub(crate) enable_webtransport: bool,
+    pub(crate) grease: bool,
+    pub(crate) max_field_section_size: u64,
+}
+
+impl Default for Params {
+    fn default() -> Self {
+        Self {
+            enable_webtransport: false,
+            grease: true,
+            max_field_section_size: Self::DEFAULT_MAX_FIELD_SECTION_SIZE,
+        }
+    }
+}
+
+impl Params {
+    /// Default max header size
+    pub const DEFAULT_MAX_FIELD_SECTION_SIZE: u64 = (1 << 62) - 1;
+
+    /// Set whether WebTransport is supported
+    pub fn enable_webtransport(mut self, val: bool) -> Self {
```

Comment:
It might be better to put the WebTransport stuff behind a Feature Flag. What do you think?

Reply:
Yeah, I thought about that too. But what if it's just for params and settings?

Reply:
I would put WebTransport completely behind a feature flag or maybe even in a separate crate. Iirc there was some discussion about this in #71.

Reply:
I understand the benefits of putting stuff behind a feature gate sometimes (smaller binary, faster compilation, etc). I totally agree we should feature-gate or separate major implementations of WT.

But configs and settings are lightweight and hard to separate. The benefit of putting those behind a feature gate seems limited. I'm 100% sure about this though. What do you think? @seanmonstar

Reply:
I have not had the time to read the WebTransport spec, so not sure what more is needed, but: does adding this at least allow for WebTransport to be _proxied_? If it's needed for a proxy to simply pass the data frames along, then seems fair to include it as part of the crate, perhaps?

Reply:
Yeah, I think so. The setting would be required to be sent by both sides when establishing connection to create a WT session.